cirq/ops/raw_types.py

```diff
@@ -371,6 +379,14 @@ def controlled_by(self,
 
         Args:
             control_qubits: Qubits to control the operation by. Required.
+            control_values: For which control qubit values to apply the
+                operation.  A sequence of the same length as `control_qubits`
+                where each entry is an integer (or set of integers)
+                corresponding to the qubit value (or set of possible values)
+                where that control is enabled.  When all controls are enabled,
+                the operation is applied.  If unspecified, control values
+                default to 1.
+            TODO: add control_qid_shape
```

Comment:
Do the todo.

Reply:
Given that I reverted all changes to Operation and control protocol from this PR, I figured it would be better to address in a subsequent one along with the control protocol changes.
But if we want to delete control protocol as per the comment below, I'll do this TODO here itself.